### ISSUE TITLE
Support nested repeating arrays

### ISSUE THREAD

**Issue by rick-nu** [particle-php/Filter#39]
### What

When there is a nested repeating array, you would like to filter that data without having to pull your data array apart. With an each function (like Particle Validator has) these could be filtered with the primary filter object.
### Expected

``` php
<?php
use Particle\Filter\Filter;

$filter = new Filter;

$filter->value('results')->each(function(Filter $filter) {
    $filter->value('vote')->bool();
});

$result = $filter->filter([
    'name' => 'Vote list',
    'votes' => [
        ['vote' => 'yes'],
        ['vote' => 'no'],
        ['vote' => 'yes'],
    ],
]);

var_dump($result);
/**
 * array(2) {
 *     ["name"]=> string(12) "Vote list"
 *     ["votes"]=> array(3) {
 *         array(1) { ["vote"]=> bool(true) },
 *         array(1) { ["vote"]=> bool(false) },
 *         array(1) { ["vote"]=> bool(true) },
 *     }
 * } 
 */
```

**Comment by rskuipers** [comment]
What do you think of introducing the following method?

``` php
$this->filter->value('votes.*.vote')->bool();
```

**Comment by rick-nu** [comment]
Implemented each functionality in #44

Discussed with @rskuipers that allowing his request would add a lot of complexity to the library itself and would probably require new dependencies to implement it properly. The each rule is an easy and stable alternative.

**Comment by rick-nu** [comment]
This feature will be released with version 2
